platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/TestResult.java

```diff
@@ -93,9 +94,12 @@ public void addFailure(SerializableFailure failure) {
         failures.add(failure);
     }
 
-    public void setIgnored() {
+    public void markIgnored(@Nullable SerializableFailure assumptionFailure) {
```

Comment:
🤔 Rather than a method with a nullable arg, would it make sense to keep 2 separate methods, a no-arg `markIgnored()` and a 1 non-null-arg `markSkipped(SerializableFailure)`?

Reply:
This would just move the null check around so the other code would change from:
```
result.markSkipped(testResult.assumptionFailure)
```
to
```
if (testResult.assumptionFailure == null) {
   result.markSkipped()
} else {
   result.markSkipped(testResult.assumptionFailure)
}
```
There's a single place this is called, so it doesn't make much of a difference to me.